master/pkg/stream/stream.go

```diff
@@ -171,6 +191,51 @@ func (p *Publisher[T]) CloseAllStreamers() {
 	p.Subscriptions = nil
 }
 
+// hydrateMsg queries the DB by the ID from rawMsg of a upsert or fallin event
+// and grabs the fields(hydrated message) that we care about.
+// Here are the different scenarios of an event in this function:
+// 1. It checks if the event has been deleted.
+// 2. It checks if the event still has the same Seq as the rawMsg.
+// 3. It checks if the event has a Seq greater than the rawMsg.
+// 4. It checks if the event is now considered as a fallout event for the subscriber and still has the same Seq as the rawMsg.
+// 5. It checks if the event is now considered as a fallout event for the subscriber and has a Seq greater than the rawMsg.
+// The function returns an upsert message scenarios 2 and a delete message in scenario 4.
+func (p *Publisher[T]) hydrateMsg(rawMsg T, idToEventCache map[int]EventCache, sub *Subscription[T], ev Event[T]) interface{} {
+	var msg interface{}
+	eventCache := EventCache{}
+
+	hydratedMsg, err := sub.hydrator(rawMsg.GetID())
+	if err != nil && errors.Cause(err) == sql.ErrNoRows {
+		// This id has a delete event later
+		eventCache.Seq = -1
+		eventCache.DeleteCache = sub.Streamer.PrepareFn(hydratedMsg.DeleteMsg())
+		idToEventCache[rawMsg.GetID()] = eventCache
+		return nil
+	} else if err != nil {
+		log.Debugf("failed to hydrate message: %s", err.Error())
```

Comment:
should the result of this error be more violent? wouldn't this cause strange inconsistencies in the clients.

Reply:
Errorf(network error or db error)